docs/src/pages/demos/autocomplete/IntegrationReactSelect.tsx

```diff
@@ -0,0 +1,290 @@
+/* eslint-disable no-undef, import/named, react/jsx-filename-extension */
+/* eslint-disable react/jsx-no-comment-textnodes, react/jsx-handler-names */
+
+import React, { CSSProperties, HTMLAttributes } from 'react';
+import clsx from 'clsx';
+import Select from 'react-select';
+import { StyleRules, Theme, WithStyles, withStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+import NoSsr from '@material-ui/core/NoSsr';
+import TextField, { BaseTextFieldProps } from '@material-ui/core/TextField';
+import Paper from '@material-ui/core/Paper';
+import Chip from '@material-ui/core/Chip';
+import MenuItem from '@material-ui/core/MenuItem';
+import CancelIcon from '@material-ui/icons/Cancel';
+import { emphasize } from '@material-ui/core/styles/colorManipulator';
+import { ValueContainerProps } from 'react-select/lib/components/containers';
+import { ControlProps } from 'react-select/lib/components/Control';
+import { MenuProps, NoticeProps } from 'react-select/lib/components/Menu';
+import { MultiValueProps } from 'react-select/lib/components/MultiValue';
+import { OptionProps } from 'react-select/lib/components/Option';
+import { PlaceholderProps } from 'react-select/lib/components/Placeholder';
+import { SingleValueProps } from 'react-select/lib/components/SingleValue';
+import { ValueType } from 'react-select/lib/types';
+
+interface OptionType {
+  label: string;
+  value: string;
+}
+
+const suggestions: OptionType[] = [
+  { label: 'Afghanistan' },
+  { label: 'Aland Islands' },
+  { label: 'Albania' },
+  { label: 'Algeria' },
+  { label: 'American Samoa' },
+  { label: 'Andorra' },
+  { label: 'Angola' },
+  { label: 'Anguilla' },
+  { label: 'Antarctica' },
+  { label: 'Antigua and Barbuda' },
+  { label: 'Argentina' },
+  { label: 'Armenia' },
+  { label: 'Aruba' },
+  { label: 'Australia' },
+  { label: 'Austria' },
+  { label: 'Azerbaijan' },
+  { label: 'Bahamas' },
+  { label: 'Bahrain' },
+  { label: 'Bangladesh' },
+  { label: 'Barbados' },
+  { label: 'Belarus' },
+  { label: 'Belgium' },
+  { label: 'Belize' },
+  { label: 'Benin' },
+  { label: 'Bermuda' },
+  { label: 'Bhutan' },
+  { label: 'Bolivia, Plurinational State of' },
+  { label: 'Bonaire, Sint Eustatius and Saba' },
+  { label: 'Bosnia and Herzegovina' },
+  { label: 'Botswana' },
+  { label: 'Bouvet Island' },
+  { label: 'Brazil' },
+  { label: 'British Indian Ocean Territory' },
+  { label: 'Brunei Darussalam' },
+].map(suggestion => ({
+  value: suggestion.label,
+  label: suggestion.label,
+}));
+
+const styles = (theme: Theme): StyleRules => ({
```

Comment:
We don't recommend using this. `createStyles` should be used until const literals land in typescript.